### ISSUE TITLE
read_csv() crashes R when file is locked by another process

### ISSUE THREAD

**Issue by vijayrajesh98** [tidyverse/readr#395]
I am reading a csv file using read_csv(). It has about 3000 records and 300 columns. 

But when I have the csv file open in parallel on my excel when I read it, it crashes my RStudio session. Is it because of my setting, or an issue with read_csv()?

Here is my session info.

R version 3.1.1 (2014-07-10)
Platform: x86_64-apple-darwin10.8.0 (64-bit)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] readr_0.2.2.9000 tidyr_0.2.0      dplyr_0.4.1

**Comment by hadley** [comment]
@jimhester this is likely to be because I've forgotten to check that the file open was successful. I'd start by looking in `SourceFile.h` around here:

```cpp
    try {
      fm_ = boost::interprocess::file_mapping(path.c_str(),
        boost::interprocess::read_only);
      mr_ = boost::interprocess::mapped_region(fm_,
        boost::interprocess::read_only);
    } catch(boost::interprocess::interprocess_exception& e) {
      Rcpp::stop("Cannot read file %s", path) ;
    }
```

It's probably unique to windows since windows has different file locking behaviour.

**Comment by hadley** [comment]
We can't reproduce unfortunately

**Comment by simonthelwall** [comment]
I've just been having the same problem with win7. I had the csv open in Excel 2010. Closing Excel resolved the problem. 
I can provide the csv file if this would be useful?

Session info below. 
```r
> devtools::session_info()
Session info ---------------------------------------------------------------------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.3 (2017-11-30)
 system   i386, mingw32               
 ui       RStudio (1.1.383)           
 language (EN)                        
 collate  English_United Kingdom.1252 
 tz       Europe/London               
 date     2018-06-18                  

Packages -------------------------------------------------------------------------------------------------------------------------------------------------------------------
 package         * version    date       source                       
 assertthat        0.2.0      2017-04-11 CRAN (R 3.4.0)               
 base            * 3.4.3      2017-12-06 local                        
 bindr             0.1.1.9000 2018-03-16 Github (krlmlr/bindr@ba5328e)
 bindrcpp        * 0.2.2      2018-03-29 CRAN (R 3.4.4)               
 compiler          3.4.3      2017-12-06 local                        
 curl              3.1        2017-12-12 CRAN (R 3.4.3)               
 datasets        * 3.4.3      2017-12-06 local                        
 devtools          1.13.5     2018-02-18 CRAN (R 3.4.3)               
 digest            0.6.15     2018-01-28 CRAN (R 3.4.3)               
 dplyr           * 0.7.5      2018-05-19 CRAN (R 3.4.4)               
 DT                0.4        2018-01-30 CRAN (R 3.4.3)               
 fingertipsR     * 0.1.7      2018-05-25 CRAN (R 3.4.4)               
 glue              1.2.0      2017-10-29 CRAN (R 3.4.3)               
 graphics        * 3.4.3      2017-12-06 local                        
 grDevices       * 3.4.3      2017-12-06 local                        
 hcaidcs         * 0.0.0.9000 2018-06-13 local                        
 hms               0.4.1      2018-01-24 CRAN (R 3.4.3)               
 htmltools         0.3.6      2017-04-28 CRAN (R 3.4.0)               
 htmlwidgets       1.0        2018-01-20 CRAN (R 3.4.3)               
 httpuv            1.3.6.2    2018-03-02 CRAN (R 3.4.1)               
 httr              1.3.1      2017-08-20 CRAN (R 3.4.1)               
 janitor         * 0.3.1      2018-01-04 CRAN (R 3.4.3)               
 jsonlite          1.5        2017-06-01 CRAN (R 3.4.0)               
 lubridate       * 1.7.3      2018-02-27 CRAN (R 3.4.1)               
 magrittr          1.5        2014-11-22 CRAN (R 3.4.0)               
 memoise           1.1.0      2017-04-21 CRAN (R 3.4.0)               
 methods         * 3.4.3      2017-12-06 local                        
 mime              0.5        2016-07-07 CRAN (R 3.4.0)               
 miniUI            0.1.1      2016-01-15 CRAN (R 3.4.1)               
 nicethings      * 0.0.0.9000 2018-06-01 local                        
 openxlsx        * 4.0.17     2017-03-23 CRAN (R 3.4.0)               
 pillar            1.2.1      2018-02-27 CRAN (R 3.4.3)               
 pkgconfig         2.0.1      2017-03-21 CRAN (R 3.4.0)               
 purrr           * 0.2.4      2017-10-18 CRAN (R 3.4.3)               
 R6                2.2.2      2017-06-17 CRAN (R 3.4.0)               
 Rcpp              0.12.16    2018-03-13 CRAN (R 3.4.4)               
 RcppRoll        * 0.2.2      2015-04-05 CRAN (R 3.4.0)               
 readr           * 1.1.1      2017-05-16 CRAN (R 3.4.0)               
 rlang             0.2.0      2018-02-20 CRAN (R 3.4.3)               
 RODBC           * 1.3-15     2017-04-13 CRAN (R 3.4.0)               
 RODBCext        * 0.3.1      2017-07-31 CRAN (R 3.4.1)               
 settings          0.2.4      2015-10-27 CRAN (R 3.4.0)               
 shiny             1.0.5      2017-08-23 CRAN (R 3.4.1)               
 shinycssloaders   0.2.0      2017-05-12 CRAN (R 3.4.1)               
 stats           * 3.4.3      2017-12-06 local                        
 stringi           1.1.7      2018-03-12 CRAN (R 3.4.4)               
 stringr         * 1.3.1      2018-05-10 CRAN (R 3.4.4)               
 tibble            1.4.2      2018-01-22 CRAN (R 3.4.3)               
 tidyr           * 0.8.1      2018-05-18 CRAN (R 3.4.4)               
 tidyselect        0.2.4      2018-02-26 CRAN (R 3.4.3)               
 tools             3.4.3      2017-12-06 local                        
 utils           * 3.4.3      2017-12-06 local                        
 validate        * 0.2.0      2017-08-07 CRAN (R 3.4.1)               
 withr             2.1.1      2017-12-19 CRAN (R 3.4.3)               
 xtable            1.8-2      2016-02-05 CRAN (R 3.4.0)               
 yaml              2.1.17     2018-02-27 CRAN (R 3.4.3)       
```

**Comment by e-clin** [comment]
I encountered this problem today in RStudio 1.3.1056 on Windows 10. I had my spreadsheet open in Excel, and when I tried to read it in using `read_csv()`, the R session was aborted. At first I suspected git, but `read_csv()` still crashed a fresh R session not in a project. I then found this post, and closing the spreadsheet in Excel solved the problem. My session info is below:

```
- Session info -----------------------------------------------------------------------------------
 setting  value                       
 version  R version 4.0.2 (2020-06-22)
 os       Windows 10 x64              
 system   x86_64, mingw32             
 ui       RStudio                     
 language (EN)                        
 collate  English_United Kingdom.1252 
 ctype    English_United Kingdom.1252 
 tz       Europe/London               
 date     2020-08-06                  

- Packages ---------------------------------------------------------------------------------------
 package     * version date       lib source        
 assertthat    0.2.1   2019-03-21 [1] CRAN (R 4.0.0)
 backports     1.1.8   2020-06-17 [1] CRAN (R 4.0.0)
 callr         3.4.3   2020-03-28 [1] CRAN (R 4.0.0)
 cli           2.0.2   2020-02-28 [1] CRAN (R 4.0.0)
 crayon        1.3.4   2017-09-16 [1] CRAN (R 4.0.0)
 desc          1.2.0   2018-05-01 [1] CRAN (R 4.0.0)
 devtools      2.3.1   2020-07-21 [1] CRAN (R 4.0.2)
 digest        0.6.25  2020-02-23 [1] CRAN (R 4.0.0)
 ellipsis      0.3.1   2020-05-15 [1] CRAN (R 4.0.0)
 fansi         0.4.1   2020-01-08 [1] CRAN (R 4.0.0)
 fs            1.5.0   2020-07-31 [1] CRAN (R 4.0.2)
 glue          1.4.1   2020-05-13 [1] CRAN (R 4.0.0)
 hms           0.5.3   2020-01-08 [1] CRAN (R 4.0.0)
 lifecycle     0.2.0   2020-03-06 [1] CRAN (R 4.0.0)
 magrittr      1.5     2014-11-22 [1] CRAN (R 4.0.0)
 memoise       1.1.0   2017-04-21 [1] CRAN (R 4.0.0)
 pillar        1.4.6   2020-07-10 [1] CRAN (R 4.0.2)
 pkgbuild      1.1.0   2020-07-13 [1] CRAN (R 4.0.2)
 pkgconfig     2.0.3   2019-09-22 [1] CRAN (R 4.0.0)
 pkgload       1.1.0   2020-05-29 [1] CRAN (R 4.0.2)
 prettyunits   1.1.1   2020-01-24 [1] CRAN (R 4.0.0)
 processx      3.4.3   2020-07-05 [1] CRAN (R 4.0.2)
 ps            1.3.3   2020-05-08 [1] CRAN (R 4.0.0)
 R6            2.4.1   2019-11-12 [1] CRAN (R 4.0.0)
 Rcpp          1.0.5   2020-07-06 [1] CRAN (R 4.0.2)
 readr         1.3.1   2018-12-21 [1] CRAN (R 4.0.0)
 remotes       2.2.0   2020-07-21 [1] CRAN (R 4.0.2)
 rlang         0.4.7   2020-07-09 [1] CRAN (R 4.0.2)
 rprojroot     1.3-2   2018-01-03 [1] CRAN (R 4.0.0)
 rstudioapi    0.11    2020-02-07 [1] CRAN (R 4.0.0)
 sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 4.0.0)
 testthat      2.3.2   2020-03-02 [1] CRAN (R 4.0.0)
 tibble        3.0.3   2020-07-10 [1] CRAN (R 4.0.2)
 tinytex       0.25    2020-07-24 [1] CRAN (R 4.0.2)
 usethis       1.6.1   2020-04-29 [1] CRAN (R 4.0.0)
 vctrs         0.3.2   2020-07-15 [1] CRAN (R 4.0.2)
 withr         2.2.0   2020-04-20 [1] CRAN (R 4.0.0)
 xfun          0.16    2020-07-24 [1] CRAN (R 4.0.2)
```

**Comment by jimhester** [comment]
Fixed by #1172
